ansible_runner/config/_base.py

```diff
@@ -193,6 +193,18 @@ def _prepare_env(self, runner_mode='pexpect'):
             self.env = {}
             # Special flags to convey info to entrypoint or process in container
             self.env['LAUNCHED_BY_RUNNER'] = '1'
+
+            if self.process_isolation_executable == 'podman':
+                # A kernel bug in RHEL < 8.5 causes podman to use the fuse-overlayfs driver. This results in errors when
```

Comment:
Since this is a RHEL < 8.5 issue, should there be a conditional check when to enable ANSIBLE_UNSAFE_WRITES? I'm assuming if a user is using RHEL8.5+ you'd want to continue using the original behavior. 

And to add to that, would there be a plan to look at other distros regarding that conditional check?

Reply:
That assumption implies that we would know whether people who have upgraded have changed their storage driver.  Which may also require that they have reset all prior podman content.  Also, since 8.5 hasn't been released and the blog post says:

> For the RHEL/CentOS Stream, we plan on backporting the feature for the RHEL8.5 release in the fall.

It would probably be wise to ensure the backport does indeed make it into 8.5 first.

Reply:
What do you all think about checking the values of NAME and VERSION_ID in `/etc/os-release`? It would be nice have this auto-disable when 8.5 is out.

Reply:
This is really a kernel bug that will be fixed in RHEL 8.5. From [this article](https://www.redhat.com/sysadmin/podman-rootless-overlay) it sounds like Linux >= 5.11 _without_ SELinux or >=5.13 _with_ SELinux will not have this problem. 

> The kernel gained rootless support in the 5.11 kernel, but a bug prevented SELinux use with the file system; this bug was fixed in 5.13.

> As far as other distributions are concerned, this support will show up when kernel 5.13 is released. For the RHEL/CentOS Stream, we plan on backporting the feature for the RHEL8.5 release in the fall.

Reply:
I think ideally, the check should check for functionality instead of a version.  `/etc/containers/storage.conf` sets the driver to `fuse-overlayfs`.  So I believe someone would have to change that still after the upgrade.  Additionally, users could configure `fuse-overlayfs` as the default even afterwards.  It might require running `podman info` and inspecting `store.graphDriverName` or `store.graphOptions.overlay.mount_program.Executable`

Reply:
We currently only support EEs that are based on UBI, so I think we can make _some_ assumptions about the environment for this scenario. 

Since this only affects local copy operations inside the EE, and will not affect remote host.. I dont care all that much. If we want to move forward with things as-is, we can remove it when our EEs are based on UBI 8.5.

Reply:
Identifying the distribution got a lot more complicated since `platform.linux_distribution()` was removed in Python 3.7. The `distro` library is a good alternative, which Ansible Core vendors. But there isn't anything equivalent available to Ansible Runner, so accurately identifying the distribution and version information, then making decisions based on that would bring with it a lot of added complexity for very little gain.

Reply:
Feature probing would be far more accurate than relying on distribution information. But I still don't think it's worth the extra effort, and even feature probing may prove unreliable. I prefer a simpler and more reliable solution.

Reply:
> We currently only support EEs that are based on UBI, so I think we can make some assumptions about the environment for this scenario.

Isn't the issue the *host* OS, not the RHEL 8 version in the EE?

Reply:
Yeah, good point, not sure what I was thinking there. 

I think we can just set this for all podman invocations until we're only supporting 8.5+ for control plane and execution nodes. So this PR LGTM.